examples/ai-core/src/e2e/feature-test-suite.ts

```diff
@@ -676,6 +677,188 @@ export function createFeatureTestSuite({
                   expect((await result.usage).totalTokens).toBeGreaterThan(0);
                 }
               });
+
+              it('should handle multiple sequential tool calls', async () => {
+                let calculatorCalls = 0;
+                const result = await generateText({
+                  model,
+                  prompt:
+                    'First calculate 2+2, then calculate the result times 3 using the calculator tool.',
+                  tools: {
+                    calculator: {
+                      parameters: z.object({
+                        expression: z.string(),
+                      }),
+                      execute: async ({ expression }) => {
+                        calculatorCalls++;
+                        return String(eval(expression));
+                      },
+                    },
+                  },
+                  maxSteps: 10,
+                });
```

Comment:
Concern: the model might just calculate the result independent of the tools. Maybe there is a better tool combination, e.g. a weather tool and a music player tool "play music that fits the mood for the current weather in San Francisco" (so the model cannot just "know" the result)